function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautResponseWriter.java

```diff
@@ -112,8 +115,20 @@ public AwsProxyResponse writeResponse(
                     status + " " +
                             Response.Status.fromStatusCode(status.getCode()).getReasonPhrase());
         }
-
+        awsProxyResponse.setHeaders(getHeaders(awsProxyResponse));
         Timer.stop(TIMER_NAME);
         return awsProxyResponse;
     }
+
+    private static Map<String, String> getHeaders(AwsProxyResponse awsProxyResponse) {
+        Headers multiValueHeaders = awsProxyResponse.getMultiValueHeaders();
+        Map<String, String> headers = new HashMap<>();
+        for (Map.Entry<String, List<String>> entry : multiValueHeaders.entrySet()) {
+            List<String> headerValues = entry.getValue();
+            if (headerValues != null && headerValues.size() == 1) {
```

Comment:
Use `iterator.hasNext()` and `iterator.next()` instead of calling `size()` and `get(0)` which is less efficient

Reply:
**check**: No headers will be added if the `size > 1`, is this intended?

**suggestion**: Try this instead `headerValues.stream().first().ifPresent(value -> headers.put(entry.getKey(), value));`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-aws/pull/1338/commits/52b852ee168d8bf9c33e1f390dbffede3966f210